modal-go/cls.go

```diff
@@ -476,14 +490,41 @@ func buildFunctionOptionsProto(options *serviceOptions) (*pb.FunctionOptions, er
 
 	builder := pb.FunctionOptions_builder{}
 
-	if options.cpu != nil || options.memory != nil || options.gpu != nil {
+	if options.cpu != nil || options.cpuLimit != nil || options.memory != nil || options.memoryLimit != nil || options.gpu != nil {
 		resBuilder := pb.Resources_builder{}
+
+		if options.cpu == nil && options.cpuLimit != nil {
+			return nil, fmt.Errorf("must also specify non-zero CPU request when CPULimit is specified")
+		}
```

Comment:
This is kind of annoying. Is it a backend limitation? I think it's only true in Python since we were trying to avoid parameter proliferation , but we're already adding a new parameter here.